src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,220 @@
+package duke;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+import duke.task.Todo;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.lang.StringBuilder;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.Files;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+/**
+ * The `Storage` class represents a tool to store tasks in file and read from file.
+ * It provides methods to store tasks in file and read from file.
+ */
+public class Storage {
+
+    private final String filePath;
+
+    /**
+     * Constructor of a storage that stores items at a specified file path.
+     *
+     * @param filePath File path to store items.
+     */
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads stored tasks from the specified file path.
+     *
+     * @return A list of tasks stored in the specified file path.
+     * @throws IllegalArgumentException If error in accessing file to retrieve tasks.
+     */
+    public List<Task> load() throws DukeException {
+        List<Task> items = new ArrayList<>();
+        String home = System.getProperty("user.dir");
+        String[] splitedFile = this.filePath.split("/");
+
+        StringBuilder sb = new StringBuilder();
+        sb.append(home);
+
+        // Ensure that the file exist
+        for (int i = 0; i < splitedFile.length; i++) {
+            sb.append("/" + splitedFile[i]);
+            Path path = Paths.get(sb.toString());
+
+            // Folder not exist
+            if (i != splitedFile.length - 1 && !Files.exists(path)) {
+                try {
+                    Files.createDirectory(path);
+                } catch (Exception e) {
+                    throw new DukeException(e.getMessage());
+                }
+            }
+
+            // Folder exist but not folder
+            if (i != splitedFile.length - 1 && !Files.isDirectory(path)) {
+                throw new DukeException();
+            }
+
+            // File not exist
+            if (i == splitedFile.length - 1 && !Files.exists(path)) {
+                try {
+                    Files.createFile(path);
```

Comment:
Why did you create a for loop just to check if each directory exists? I think that System.getProperty("user.dir") already ensures that the path it returns all exists

Reply:
My concern is regarding the file path given by user, e.g. for a file path `folderA/folderB/fileC`, indeed System.getProperty("user.dir") already ensures that the path it returns all exists, but when I append the file path given by the user, it could happen certain errors such as:

1. folderA and / or folderB does not exist or exist as a file with same name
2. fileC does not exist or exist as a folder with same name

Therefore, I decide to handle this potential problem by performing checks on each layer, and create a file at the given file path iff there is no error as mentioned above.

Reply:
You could just append the path to user.dir and then check use File.exists() method once to see it's valid. A for loop is most likely unnecessary here